From b8969a1b69672b163d057e7745ebc915df689211 Mon Sep 17 00:00:00 2001
From: Ondrej Mosnacek <omosnace@redhat.com>
Date: Tue, 2 May 2023 10:02:33 +0200
Subject: crypto: api - Fix CRYPTO_USER checks for report function

Checking the config via ifdef incorrectly compiles out the report
functions when CRYPTO_USER is set to =m. Fix it by using IS_ENABLED()
instead.

Fixes: c0f9e01dd266 ("crypto: api - Check CRYPTO_USER instead of NET for report")
Signed-off-by: Ondrej Mosnacek <omosnace@redhat.com>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
---
 crypto/acompress.c | 2 +-
 crypto/aead.c      | 2 +-
 crypto/ahash.c     | 2 +-
 crypto/akcipher.c  | 2 +-
 crypto/kpp.c       | 2 +-
 crypto/rng.c       | 2 +-
 crypto/scompress.c | 2 +-
 crypto/shash.c     | 2 +-
 crypto/skcipher.c  | 2 +-
 9 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/crypto/acompress.c b/crypto/acompress.c
index 82a290df2822a..1c682810a484d 100644
--- a/crypto/acompress.c
+++ b/crypto/acompress.c
@@ -125,7 +125,7 @@ static const struct crypto_type crypto_acomp_type = {
 #ifdef CONFIG_PROC_FS
 	.show = crypto_acomp_show,
 #endif
-#ifdef CONFIG_CRYPTO_USER
+#if IS_ENABLED(CONFIG_CRYPTO_USER)
 	.report = crypto_acomp_report,
 #endif
 #ifdef CONFIG_CRYPTO_STATS
diff --git a/crypto/aead.c b/crypto/aead.c
index ffc48a7dfb349..d5ba204ebdbfa 100644
--- a/crypto/aead.c
+++ b/crypto/aead.c
@@ -242,7 +242,7 @@ static const struct crypto_type crypto_aead_type = {
 #ifdef CONFIG_PROC_FS
 	.show = crypto_aead_show,
 #endif
-#ifdef CONFIG_CRYPTO_USER
+#if IS_ENABLED(CONFIG_CRYPTO_USER)
 	.report = crypto_aead_report,
 #endif
 #ifdef CONFIG_CRYPTO_STATS
diff --git a/crypto/ahash.c b/crypto/ahash.c
index b8a607928e72d..3246510404465 100644
--- a/crypto/ahash.c
+++ b/crypto/ahash.c
@@ -509,7 +509,7 @@ static const struct crypto_type crypto_ahash_type = {
 #ifdef CONFIG_PROC_FS
 	.show = crypto_ahash_show,
 #endif
-#ifdef CONFIG_CRYPTO_USER
+#if IS_ENABLED(CONFIG_CRYPTO_USER)
 	.report = crypto_ahash_report,
 #endif
 #ifdef CONFIG_CRYPTO_STATS
diff --git a/crypto/akcipher.c b/crypto/akcipher.c
index 186e762b509a6..7960ceb528c36 100644
--- a/crypto/akcipher.c
+++ b/crypto/akcipher.c
@@ -98,7 +98,7 @@ static const struct crypto_type crypto_akcipher_type = {
 #ifdef CONFIG_PROC_FS
 	.show = crypto_akcipher_show,
 #endif
-#ifdef CONFIG_CRYPTO_USER
+#if IS_ENABLED(CONFIG_CRYPTO_USER)
 	.report = crypto_akcipher_report,
 #endif
 #ifdef CONFIG_CRYPTO_STATS
diff --git a/crypto/kpp.c b/crypto/kpp.c
index 74f2e8e918fa5..33d44e59387ff 100644
--- a/crypto/kpp.c
+++ b/crypto/kpp.c
@@ -96,7 +96,7 @@ static const struct crypto_type crypto_kpp_type = {
 #ifdef CONFIG_PROC_FS
 	.show = crypto_kpp_show,
 #endif
-#ifdef CONFIG_CRYPTO_USER
+#if IS_ENABLED(CONFIG_CRYPTO_USER)
 	.report = crypto_kpp_report,
 #endif
 #ifdef CONFIG_CRYPTO_STATS
diff --git a/crypto/rng.c b/crypto/rng.c
index ffde0f64fb259..279dffdebf598 100644
--- a/crypto/rng.c
+++ b/crypto/rng.c
@@ -118,7 +118,7 @@ static const struct crypto_type crypto_rng_type = {
 #ifdef CONFIG_PROC_FS
 	.show = crypto_rng_show,
 #endif
-#ifdef CONFIG_CRYPTO_USER
+#if IS_ENABLED(CONFIG_CRYPTO_USER)
 	.report = crypto_rng_report,
 #endif
 #ifdef CONFIG_CRYPTO_STATS
diff --git a/crypto/scompress.c b/crypto/scompress.c
index 24138b42a648a..442a82c9de7de 100644
--- a/crypto/scompress.c
+++ b/crypto/scompress.c
@@ -240,7 +240,7 @@ static const struct crypto_type crypto_scomp_type = {
 #ifdef CONFIG_PROC_FS
 	.show = crypto_scomp_show,
 #endif
-#ifdef CONFIG_CRYPTO_USER
+#if IS_ENABLED(CONFIG_CRYPTO_USER)
 	.report = crypto_scomp_report,
 #endif
 #ifdef CONFIG_CRYPTO_STATS
diff --git a/crypto/shash.c b/crypto/shash.c
index 5845b7d59b2f2..717b42df3495e 100644
--- a/crypto/shash.c
+++ b/crypto/shash.c
@@ -548,7 +548,7 @@ static const struct crypto_type crypto_shash_type = {
 #ifdef CONFIG_PROC_FS
 	.show = crypto_shash_show,
 #endif
-#ifdef CONFIG_CRYPTO_USER
+#if IS_ENABLED(CONFIG_CRYPTO_USER)
 	.report = crypto_shash_report,
 #endif
 #ifdef CONFIG_CRYPTO_STATS
diff --git a/crypto/skcipher.c b/crypto/skcipher.c
index 6caca02d7e552..7b275716cf4e3 100644
--- a/crypto/skcipher.c
+++ b/crypto/skcipher.c
@@ -776,7 +776,7 @@ static const struct crypto_type crypto_skcipher_type = {
 #ifdef CONFIG_PROC_FS
 	.show = crypto_skcipher_show,
 #endif
-#ifdef CONFIG_CRYPTO_USER
+#if IS_ENABLED(CONFIG_CRYPTO_USER)
 	.report = crypto_skcipher_report,
 #endif
 #ifdef CONFIG_CRYPTO_STATS
-- 
cgit 

